server/game/cards/08.1-TAK/BloodyArakh.js

```diff
@@ -0,0 +1,36 @@
+const DrawCard = require('../../drawcard.js');
+
+class BloodyArakh extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            max: ability.limit.perPhase(1),
+            when: {
+                afterChallenge: event => event.challenge.winner === this.controller && event.challenge.challengeType === 'military' &&
```

Comment:
Unless I'm missing something Bloody Arakh can be attached to opponent characters and, in that case, will need to trigger when the opponent wins a challenge, not the attachment controller. So this check needs to be a bit more complicated, unfortunately.